plugins/modules/ios_route_maps.py

```diff
@@ -0,0 +1,2357 @@
+                  connected:
+                    description: Connected
+                    type: bool
+                  eigrp:
+                    description:
+                      - Enhanced Interior Gateway Routing Protocol (EIGRP)
+                      - Autonomous system number
+                      - Please refer vendor documentation for valid values
+                    type: int
+                  isis:
+                    description: ISO IS-IS
+                    type: bool
+                  lisp:
+                    description: Locator ID Separation Protocol (LISP)
+                    type: bool
+                  mobile:
+                    description: Mobile routes
+                    type: bool
+                  ospf:
+                    description:
+                      - Open Shortest Path First (OSPF) Process ID
+                      - Please refer vendor documentation for valid values
+                    type: int
+                  ospfv3:
+                    description:
+                      - OSPFv3 Process ID
+                      - Please refer vendor documentation for valid values
+                    type: int
+                  rip:
+                    description: Routing Information Protocol (RIP)
+                    type: bool
+                  static:
+                    description: Static routes
+                    type: bool
+              tag:
+                description: Match tag of route
+                type: dict
+                suboptions:
+                  value:
+                    description: Tag value/Tag in Dotted Decimal eg, 10.10.10.10
+                    type: list
+                    elements: str
+                  tag_list:
+                    description: Route Tag List/Tag list name
+                    type: list
+                    elements: str
+              track:
+                description: tracking object
+                type: int
+          set:
+            description: Match source-protocol of route
+            type: dict
+            suboptions:
+              aigp_metric:
+                description: accumulated metric value
+                type: dict
+                suboptions:
+                  value:
+                    description: manual value
+                    type: int
+                  igp_metric:
+                    description: metric value from rib
+                    type: bool
+              as_path:
+                description: Prepend string for a BGP AS-path attribute
+                type: dict
+                suboptions:
+                  prepend:
+                    description: Prepend to the as-path
+                    type: dict
+                    suboptions:
+                      as_number:
+                        description:
+                          - AS number
+                          - Please refer vendor documentation for valid values
+                        type: str
```

Comment:
There can be multiple `as_number` entries. 
```suggestion
                        type: list
                        elements: str
```